BGL/include/CGAL/boost/graph/shortest_path.h

```diff
@@ -0,0 +1,190 @@
+
+  public:
+    vertex_descriptor destination_vd;
+    VertexEdgeMap& relaxed_edges;
+
+    Stop_at_target_Dijkstra_visitor(vertex_descriptor destination_vd,
+                                    VertexEdgeMap& relaxed_edges)
+      : destination_vd(destination_vd), relaxed_edges(relaxed_edges)
+    {}
+
+    void initialize_vertex(const vertex_descriptor& /*s*/, const Graph& /*g*/) const {}
+    void examine_vertex(const vertex_descriptor& /*s*/, const Graph& /*g*/) const {}
+    void examine_edge(const edge_descriptor& /*e*/, const Graph& /*g*/) const {}
+    void edge_relaxed(const edge_descriptor& e, const Graph& g) const
+    {
+      relaxed_edges[target(e, g)] = e;
+    }
+    void discover_vertex(const vertex_descriptor& /*s*/, const Graph& /*g*/) const {}
+    void edge_not_relaxed(const edge_descriptor& /*e*/, const Graph& /*g*/) const {}
+    void finish_vertex(const vertex_descriptor& vd, const Graph& /* g*/) const
+    {
+      if (vd == destination_vd)
+        throw Dijkstra_end_exception();
+    }
+  };
+} // namespace internal
+
+/*!
+* \ingroup PkgBGLTraversal
+* Computes the shortest path between two vertices in a graph `g`.
+* The vertices must belong to the same connected component of `g`.
+*
+* @tparam Graph a model of the concept `HalfedgeListGraph`
+* @tparam OutputIterator an output iterator with value type `boost::graph_traits<Graph>::%halfedge_descriptor`
+* @tparam NamedParameters a sequence of \ref bgl_namedparameters "Named Parameters"
+*
+* @param vs source vertex
+* @param vt target vertex
+* @param g the graph
+* @param halfedge_sequence_oit the output iterator holding the output sequence
+* of halfedges that form the shortest path from `vs` to `vt` on `g`
+* @param np an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below
+*
+*  \cgalNamedParamsBegin
+*   \cgalParamNBegin{edge_weight_map}
+*    \cgalParamDescription{a property map associating to each edge in the graph its weight or ``length''.
+*                          The weights must all be non-negative.}
+*    \cgalParamType{a class model of `ReadablePropertyMap` with `boost::graph_traits<PolygonMesh>::%edge_descriptor`
+*                   as key type and `FT` as value type.}
+*    \cgalParamDefault{`get(boost::edge_weight, mesh)`}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{vertex_index_map}
+*     \cgalParamDescription{a property map associating to each vertex of `pmesh` a unique index between `0` and `num_vertices(pmesh) - 1`}
+*     \cgalParamType{a class model of `ReadablePropertyMap` with `boost::graph_traits<PolygonMesh>::%vertex_descriptor`
+*                    as key type and `std::size_t` as value type}
+*     \cgalParamDefault{an automatically indexed internal map}
+*   \cgalParamNEnd
+*  \cgalNamedParamsEnd
+*/
+template<typename Graph,
+         typename OutputIterator,
+         typename NamedParameters = parameters::Default_named_parameters>
+OutputIterator shortest_path_between_two_vertices(
```

Comment:
```suggestion
OutputIterator shortest_path(
```

Reply:
@sloriot any opinion on the function name?

Reply:
`dijkstra_shortest_path()` so that it's clear it's not geodesic